examples/pipelines/datacomp/components/dedup_image_embeddings/src/main.py

```diff
@@ -0,0 +1,96 @@
+"""This component deduplicates images based on embeddings.
+
+As proposed in [Abbas et al., 2022](https://arxiv.org/abs/2303.09540).
+
+Implementation is based on https://github.com/conceptofmind/SemDeDup/tree/main.
+"""
+import logging
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+from scipy.spatial.distance import cdist
+from sklearn.metrics.pairwise import cosine_similarity
+
+from fondant.component import DaskTransformComponent
+from fondant.executor import DaskTransformExecutor
+
+logger = logging.getLogger(__name__)
+
+
+def sort_by_centroid_distance(embeddings, centroid, descending=True):
+    distances = cdist(embeddings, centroid.reshape(1, -1), "euclidean")
+    sorted_indices = np.argsort(distances, axis=0)
+    if descending:
+        sorted_indices = sorted_indices[::-1]
+    return embeddings[sorted_indices.flatten()]
+
+
+def deduplicate(df, epsilon):
```

Comment:
Can you add types here? I assumed at first that this was a Dask dataframe, but if I understand correctly it's a pandas dataframe.